include/sound/sof/dai.h

```diff
@@ -57,7 +57,13 @@
 #define SOF_DAI_CONFIG_FLAGS_NONE	(0 << 0) /**< DAI_CONFIG sent without stage information */
 #define SOF_DAI_CONFIG_FLAGS_HW_PARAMS	(1 << 0) /**< DAI_CONFIG sent during hw_params stage */
 #define SOF_DAI_CONFIG_FLAGS_HW_FREE	(2 << 0) /**< DAI_CONFIG sent during hw_free stage */
-#define SOF_DAI_CONFIG_FLAGS_RFU	(3 << 0) /**< not used, reserved for future use */
+/*
+ * When used along with the SOF_DAI_CONFIG_FLAGS_HW_PARAMS, it indicates that pipeline stop and
+ * DAI DMA stop should happen in two steps. When used without the SOF_DAI_CONFIG_FLAGS_HW_PARAMS
+ * flag, it is used to pause the DMA after pipeline pause. This change is only available ABI 3.20
+ * onwards.
+ */
+#define SOF_DAI_CONFIG_FLAGS_2_STEP_STOP_PAUSE (3 << 0)
```

Comment:
This is getting rather complex with DAI_CONFIG flags modifying the semantics more and more. Perhaps just adding a separate DAI_UNCONFIG IPC would be clearer to build on. OTOH, we already sent the FLAGS_HW_FREE upstream, so this would not look nice. And @plbossart you explicitly cover this argument in https://github.com/thesofproject/sof/commit/7ff529055478d56c2277514133b361baac5add4e

But given this is still a sub command, I think we'd need to have a separate MASK for the DAI_CONFIG CMD (hw_params or hw_free) and then a separate mask for flags that can have different semantics. E.g. the 2-step-stop-pause is clearly a setup flag.

Reply:
right I think we need to expand the bitfields to use all 8 bits along with the mask. let me fix this